cmd/syncthing/gui.go

```diff
@@ -1024,6 +1027,93 @@ func (s *apiService) getSystemLogTxt(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+type fileEntry struct {
+	name string
+	data []byte
+}
+
+func (s *apiService) getSupportBundle(w http.ResponseWriter, r *http.Request) {
+	var files []fileEntry
+
+	// Redacted configuration as a JSON
+	if jsonConfig, err := json.MarshalIndent(getRedactedConfig(s), "", "  "); err == nil {
+		files = append(files, fileEntry{name: "config.json", data: jsonConfig})
+	}
+
+	// Log as a text
+	var buflog bytes.Buffer
+	for _, line := range s.systemLog.Since(time.Time{}) {
+		fmt.Fprintf(&buflog, "%s: %s\n", line.When.Format(time.RFC3339), line.Message)
+	}
+	files = append(files, fileEntry{name: "log.txt", data: buflog.Bytes()})
+
+	// Errors as a JSON
+	if errs := s.guiErrors.Since(time.Time{}); len(errs) > 0 {
+		if jsonError, err := json.MarshalIndent(errs, "", "  "); err == nil {
+			files = append(files, fileEntry{name: "errors.json", data: jsonError})
+		}
+	}
+
+	// Panic files as a JSON
+	if panicFiles, err := filepath.Glob(filepath.Join(baseDirs["config"], "panic*")); err == nil {
+		for _, f := range panicFiles {
+			if panicFile, err := ioutil.ReadFile(f); err == nil {
+				files = append(files, fileEntry{name: filepath.Base(f), data: panicFile})
+			}
+		}
+	}
+
+	// Version and platform information as a JSON
+	if versionPlatform, err := json.MarshalIndent(map[string]string{
+		"version":     Version,
+		"codename":    Codename,
+		"longVersion": LongVersion,
+		"os":          runtime.GOOS,
+		"arch":        runtime.GOARCH,
+	}, "", "  "); err == nil {
+		files = append(files, fileEntry{name: "versionPlatform.json", data: versionPlatform})
+	}
+
+	// Report Data as a JSON
+	if usageReportingData, err := json.MarshalIndent(reportData(s.cfg, s.model, s.connectionsService, usageReportVersion, true), "", "  "); err == nil {
+		files = append(files, fileEntry{name: "usageReportingData.json", data: usageReportingData})
+	}
+
+	// Heap and CPU Proofs as a pprof extension
+	var heapBuffer, cpuBuffer bytes.Buffer
+	filename := fmt.Sprintf("syncthing-heap-%s-%s-%s-%s.pprof", runtime.GOOS, runtime.GOARCH, Version, time.Now().Format("150405")) // hhmmss
+	runtime.GC()
+	pprof.WriteHeapProfile(&heapBuffer)
+	files = append(files, fileEntry{name: filename, data: heapBuffer.Bytes()})
+
+	const duration = 4 * time.Second
+	filename = fmt.Sprintf("syncthing-cpu-%s-%s-%s-%s.pprof", runtime.GOOS, runtime.GOARCH, Version, time.Now().Format("150405")) // hhmmss
+	pprof.StartCPUProfile(&cpuBuffer)
+	time.Sleep(duration)
+	pprof.StopCPUProfile()
+	files = append(files, fileEntry{name: filename, data: cpuBuffer.Bytes()})
+
+	// Add buffer files to buffer zip
+	var zipFilesBuffer bytes.Buffer
+	if err := zipBuffer(&zipFilesBuffer, files); err != nil {
+		l.Warnln("Support Bundle zip not created: ", err)
```

Comment:
"Failed to create support bundle zip:"

Reply:
Also, no point creating zip file etc if ti failed? Just return a 500 and be done with it?